File: API/src/Padrao.Business/Models/Imovel.cs
namespace Padrao.Business.Models;
public class Imovel : Entity
{
    public string Nome { get; set; }
    public string Descricao { get; set; }
    public int CategoriaId { get; set; }
    public int TipoImovelId { get; set; }
    public TipoImovel TipoImovel { get; set; }
    public Categoria Categoria { get; set; }
    public IEnumerable<Foto> Fotos { get; set; }

}

File: APP/Padrao-App/src/app/shared/models/imovel.ts
import { Categoria } from "./categoria";
import { Entity } from "./entity";
import { Foto } from "./foto";
import { TipoImovel } from "./tipo-imovel";


export interface Imovel extends Entity {
    nome: string;
    descricao: string;
    tipoImovel: TipoImovel;
    categoria: Categoria;
    fotos: Foto[];
}
